volume-group-operator/controllers/utils/volumegroup.go

```diff
@@ -46,3 +67,67 @@ func UpdateVolumeGroupStatusError(client client.Client, instance *volumegroupv1.
 
 	return nil
 }
+
+func GetVGList(logger logr.Logger, client client.Client) (volumegroupv1.VolumeGroupList, error) {
+	logger.Info(messages.ListVolumeGroups)
+	vg := &volumegroupv1.VolumeGroupList{}
+	err := client.List(context.TODO(), vg)
+	if err != nil {
+		return volumegroupv1.VolumeGroupList{}, err
+	}
+	return *vg, nil
+}
+
+func IsPVCMatchesVG(logger logr.Logger, client client.Client,
+	pvc *corev1.PersistentVolumeClaim, vg volumegroupv1.VolumeGroup) (bool, error) {
+
+	logger.Info(fmt.Sprintf(messages.CheckIfPersistentVolumeClaimMatchesVolumeGroup,
+		pvc.Namespace, pvc.Name, vg.Namespace, vg.Name))
+	areLabelsMatchLabelSelector, err := areLabelsMatchLabelSelector(
+		client, pvc.ObjectMeta.Labels, *vg.Spec.Source.Selector)
+
+	if areLabelsMatchLabelSelector {
+		logger.Info(fmt.Sprintf(messages.PersistentVolumeClaimMatchedToVolumeGroup,
+			pvc.Namespace, pvc.Name, vg.Namespace, vg.Name))
+		return true, err
+	} else {
+		logger.Info(fmt.Sprintf(messages.PersistentVolumeClaimNotMatchedToVolumeGroup,
+			pvc.Namespace, pvc.Name, vg.Namespace, vg.Name))
+		return false, err
+	}
+}
+
+func IsPVCPartOfVG(pvc *corev1.PersistentVolumeClaim, pvcListInVG []corev1.PersistentVolumeClaim) bool {
+	for _, pvcFromList := range pvcListInVG {
+		if pvcFromList.Name == pvc.Name && pvcFromList.Namespace == pvc.Namespace {
+			return true
+		}
+	}
+	return false
+}
+
+func RemovePVCFromVG(logger logr.Logger, client client.Client, pvc *corev1.PersistentVolumeClaim, vg *volumegroupv1.VolumeGroup) error {
+	logger.Info(fmt.Sprintf(messages.RemovePersistentVolumeClaimFromVolumeGroup,
+		pvc.Namespace, pvc.Name, vg.Namespace, vg.Name))
+	vg.Status.PVCList = removePersistentVolumeClaimFromVolumeGroupPVCList(pvc, vg.Status.PVCList)
+	err := client.Status().Update(context.TODO(), vg)
+	if err != nil {
+		logger.Error(err, fmt.Sprintf(messages.FailedToRemovePersistentVolumeClaimFromVolumeGroup,
+			pvc.Namespace, pvc.Name, vg.Namespace, vg.Name))
+		return err
+	}
+	logger.Info(fmt.Sprintf(messages.RemovedPersistentVolumeClaimFromVolumeGroup,
+		pvc.Namespace, pvc.Name, vg.Namespace, vg.Name))
+	return nil
+}
+
+func removePersistentVolumeClaimFromVolumeGroupPVCList(pvc *corev1.PersistentVolumeClaim,
+	pvcListInVG []corev1.PersistentVolumeClaim) []corev1.PersistentVolumeClaim {
```

Comment:
in this context, I don't think it's important to know the VG
```suggestion
	pvcList []corev1.PersistentVolumeClaim) []corev1.PersistentVolumeClaim {
```

Reply:
done